test_unstructured/chunking/test_title.py

```diff
@@ -705,19 +706,130 @@ def test_it_considers_separator_length_when_sectioning():
 class Describe_NonTextSection:
     """Unit-test suite for `unstructured.chunking.title._NonTextSection objects."""
 
-    def it_provides_access_to_its_element(self):
+    def it_iterates_its_element_as_the_sole_chunk(self):
         checkbox = CheckBox()
         section = _NonTextSection(checkbox)
-        assert section.element is checkbox
+
+        chunk_iter = section.iter_chunks(maxlen=500)
+
+        chunk = next(chunk_iter)
+        assert isinstance(chunk, CheckBox)
+        with pytest.raises(StopIteration):
+            next(chunk_iter)
 
 
 class Describe_TableSection:
     """Unit-test suite for `unstructured.chunking.title._TableSection objects."""
 
-    def it_provides_access_to_its_table(self):
-        table = Table("<table></table>")
-        section = _TableSection(table)
-        assert section.table is table
+    def it_uses_its_table_as_the_sole_chunk_when_it_fits_in_the_window(self):
+        html_table = (
+            "<table>\n"
+            "<thead>\n"
+            "<tr><th>Header Col 1 </th><th>Header Col 2 </th></tr>\n"
+            "</thead>\n"
+            "<tbody>\n"
+            "<tr><td>Lorem ipsum  </td><td>adipiscing   </td></tr>\n"
+            "</tbody>\n"
+            "</table>"
+        )
+        text_table = "Header Col 1  Header Col 2\n" "Lorem ipsum   adipiscing"
+        section = _TableSection(
+            Table(text_table, metadata=ElementMetadata(text_as_html=html_table))
+        )
+
+        chunk_iter = section.iter_chunks(maxlen=175)
+
+        chunk = next(chunk_iter)
+        assert isinstance(chunk, Table)
+        assert chunk.text == "Header Col 1  Header Col 2\nLorem ipsum   adipiscing"
+        assert chunk.metadata.text_as_html == (
+            "<table>\n"
+            "<thead>\n"
+            "<tr><th>Header Col 1 </th><th>Header Col 2 </th></tr>\n"
+            "</thead>\n"
+            "<tbody>\n"
+            "<tr><td>Lorem ipsum  </td><td>adipiscing   </td></tr>\n"
+            "</tbody>\n"
+            "</table>"
+        )
+        with pytest.raises(StopIteration):
+            next(chunk_iter)
+
+    def but_it_splits_its_table_into_TableChunks_when_the_table_text_exceeds_the_window(self):
```

Comment:
Confirmed this reproduces the bug that previously existed where `text_as_html` was sought in the table element rather than the table element metadata.